### PR TITLE
Add Swift tuple initializer overlay to ORKPredicateStepNavigationRule

### PR DESCRIPTION
**Pull request by rsanchezsaez**
This adds a *Swift* overlay for `ORKPredicateStepNavigationRule` which takes an array of `(NSPredicate, String)` tuples (*result predicate*, *destination step identifier*) instead of two separate arrays. ([Issue #224](https://github.com/ResearchKit/ResearchKit/issues/224))

The *Navigable Ordered Task* creation code in `ORKTest` was converted to *Swift*, and it now uses the overlay *Swift* initializer for creating the *predicate step navigation rules*.

Note that this sets `CLANG_ENABLE_MODULES` to `YES` in the *ResearchKit* project, I hope it doesn’t introduce any unintended side-effects.

~~On `iOS 9`, we’ll be able to use [`NS_SWIFT_UNAVAILABLE(“”)`](https://developer.apple.com/library/prerelease/ios/documentation/Swift/Conceptual/BuildingCocoaApps/MixandMatch.html#//apple_ref/doc/uid/TP40014216-CH10-ID178) to disallow the raw *Objective-C* initializers from *Swift*.~~ Since we're now supporting *Xcode 7* on `master`, I used the macro to made the *ObjC* initializers unavailable from *Swift*.

Is this the correct approach for creating a *Swift* overlay?

### CODE REVIEW

**Comment by rsanchezsaez** [comment]
Hey @YuanZhu-apple, do you think this could be merged any time soon?

I'm looking at the backwards navigation problem upon `ORKNavigableOrderedTask` restoration and it would be nice to have this in to avoid conflicts.

File: Testing/ORKTest/ORKTest/TestHelpers.swift [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+
+import Foundation
+import ResearchKit
+
+@objc class TestHelpers : NSObject {
```
Comment (YuanZhu-apple) [inline Testing/ORKTest/ORKTest/TestHelpers.swift]: Is `TaskFactory` a better name for this class?

**Comment by YuanZhu-apple** [comment]
Thanks @rsanchezsaez, the Swift tuple initializer is more naturally than the Ojc version.

File: Testing/ORKTest/ORKTest/MainViewController.m [hunk 2]
```diff
@@ -475,7 +479,7 @@ - (UICollectionViewCell *)collectionView:(UICollectionView *)collectionView cell
                                                               rotated:NO
                                                             timeLimit:300.0
                                                               options:ORKPredefinedTaskOptionNone];
-    } else if ([identifier isEqualToString:StepNavigationTaskIdentifier]) {
+    } else if ([identifier isEqualToString:NavigableOrderedTaskIdentifier]) {
         return [self makeNavigableOrderedTask];
```
Comment (YuanZhu-apple) [inline Testing/ORKTest/ORKTest/MainViewController.m]: Call swift factory method directly from here?

**Comment by rsanchezsaez** [comment]
Thanks for the review, I'll make the suggested changes soon.

**Comment by rsanchezsaez** [comment]
Changes done, and synced to master. I also fixed a problem that prevented *RK's unit tests* to run due to the new embedded Swift code.

**Comment by YuanZhu-apple** [comment]
R=me.
Thanks @rsanchezsaez, merging this one.

**Comment by chrisnowak** [comment]
Hey @rsanchezsaez - I'm not 100% sure this is the root issue, but I think this new Swift overlay is not included in the ResearchKit cocoapod due to Swift files not being part of the "source" in the podspec here:
https://github.com/ResearchKit/ResearchKit/blob/master/ResearchKit.podspec#L11

Whenever I try to reference master as a cocoapod (I'm trying to play with some of the new onboarding stuff slated for next release), it can't find the "resultPredicatesAndDestinationStepIdentifiers" method.

**Pull request merged**